MobileCenter/MobileCenterTests/MSDBStorageTest.m

```diff
@@ -0,0 +1,265 @@
+#import <OCHamcrestIOS/OCHamcrestIOS.h>
+#import <OCMock/OCMock.h>
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+#import "MSDBStoragePrivate.h"
+#import "MSUtility+File.h"
+
+static NSString *const kMSTestDBFileName = @"Test.sqlite";
+static NSString *const kMSTestTableName = @"table";
+static NSString *const kMSTestPositionColName = @"position";
+static NSString *const kMSTestPersonColName = @"person";
+static NSString *const kMSTestHungrinessColName = @"hungriness";
+static NSString *const kMSTestMealColName = @"meal";
+
+@interface MSDBStorageTests : XCTestCase
+
+@property(nonatomic) MSDBStorage *sut;
+@property(nonatomic) MSDBSchema *schema;
+
+@end
+
+@implementation MSDBStorageTests
+
+- (void)setUp {
+  [super setUp];
+  self.schema = @{
+    kMSTestTableName : @[
+      @{
+        kMSTestPositionColName :
+            @[ kMSSQLiteTypeInteger, kMSSQLiteConstraintPrimaryKey, kMSSQLiteConstraintAutoincrement ]
+      },
+      @{kMSTestPersonColName : @[ kMSSQLiteTypeText, kMSSQLiteConstraintNotNull ]},
+      @{kMSTestHungrinessColName : @[ kMSSQLiteTypeInteger ]},
+      @{kMSTestMealColName : @[ kMSSQLiteTypeText, kMSSQLiteConstraintNotNull ]}
+    ]
+  };
+  self.sut = [[MSDBStorage alloc] initWithSchema:self.schema filename:kMSTestDBFileName];
+}
+
+- (void)tearDown {
+  [super tearDown];
+  [self.sut deleteDB];
+}
+
+- (void)testInitWithSchema {
+
+  // If
+  [self.sut deleteDB];
```

Comment:
It is since we don't want any schema from the tearup to interfere with the new instance created bellow.